src/plugins/console/server/routes/api/console/proxy/create_handler.ts

```diff
@@ -140,77 +85,45 @@ export const createHandler = ({
     });
   }
 
-  const legacyConfig = await readLegacyOpenSearchConfig();
-  const { hosts } = legacyConfig;
-  let opensearchIncomingMessage: IncomingMessage;
-
-  for (let idx = 0; idx < hosts.length; ++idx) {
-    const host = hosts[idx];
-    try {
-      const uri = toURL(host, path);
-
-      // Because this can technically be provided by a settings-defined proxy config, we need to
-      // preserve these property names to maintain BWC.
-      const { timeout, agent, headers, rejectUnauthorized } = getRequestConfig(
-        request.headers,
-        legacyConfig,
-        proxyConfigCollection,
-        uri.toString()
-      );
-
-      const requestHeaders = {
-        ...headers,
-        ...getProxyHeaders(request),
-      };
-
-      opensearchIncomingMessage = await proxyRequest({
-        method: method.toLowerCase() as any,
-        headers: requestHeaders,
-        uri,
-        timeout,
-        payload: body,
-        rejectUnauthorized,
-        agent,
+  try {
+    const requestHeaders = {
+      ...getProxyHeaders(request),
+    };
+
+    opensearchResponse = await client.transport.request(
+      { path: path + '?pretty=true', method, body },
+      { headers: requestHeaders }
+    );
+
+    const { statusCode, body: responseContent, warnings } = opensearchResponse;
+
+    if (method.toUpperCase() !== 'HEAD') {
+      return response.custom({
+        statusCode: statusCode!,
+        body: responseContent,
+        headers: {
+          warning: warnings || '',
+        },
       });
-
-      break;
-    } catch (e) {
-      // If we reached here it means we hit a lower level network issue than just, for e.g., a 500.
-      // We try contacting another node in that case.
-      log.error(e);
-      if (idx === hosts.length - 1) {
-        log.warn(`Could not connect to any configured OpenSearch node [${hosts.join(', ')}]`);
-        return response.customError({
-          statusCode: 502,
-          body: e,
-        });
-      }
-      // Otherwise, try the next host...
     }
-  }
-
-  const {
-    statusCode,
-    statusMessage,
-    headers: { warning },
-  } = opensearchIncomingMessage!;
 
-  if (method.toUpperCase() !== 'HEAD') {
     return response.custom({
       statusCode: statusCode!,
-      body: opensearchIncomingMessage!,
+      body: `${statusCode} - ${responseContent}`,
       headers: {
-        warning: warning || '',
+        warning: warnings || '',
+        'Content-Type': 'text/plain',
       },
     });
+  } catch (e: any) {
+    log.error(e);
+    return response.customError({
+      statusCode: isResponseError(e) ? e.statusCode : 502,
```

Comment:
How would you feel about something like this?
```suggestion
      statusCode: isResponseError(e) ? e.statusCode : `502.${e.statusCode || 0}`,
```

This will help with debugging when we see a 502 because the subtype would be what OSD itself got.